### DISCUSSION TITLE
default active.ident in Seurat object

### DISCUSSION THREAD

**Discussion by EsmaeelAzadian** [satijalab/seurat#5437] [Q&A]
Hi everyone,

Using {InstallData("bmcite"), bm <- LoadData(ds = "bmcite"), from: https://satijalab.org/seurat/articles/weighted_nearest_neighbor_analysis.html} I can see the active.ident (bm@active.ident) consists of Factor with two levels “a” and “b”. Also, each cell identity includes either “a” or “b”. Does anyone know why there is such an identity in the Seurat object? What does it mean? What is it used for? I removed “a” and “b” from cell identities and figured out the name of some cells become repetitive, perhaps it’s not a good idea to remove these identities, although when I create some graphs (e.g., FeatureScatter plot) it produces two colours for identity “a” and “b” which is confusing.

**Comment by samuel-marsh** [comment]
Hi,

If you look at the object meta data you can find that the a and b in this case correspond to the batch.  You don't want to remove from cell prefixes for exactly the reason you found.  10X barcodes are not unique between lanes and therefore there is potential for barcode name collision if not addressed (either during cell ranger processing or during creation of merged object).  

In terms of plotting you can simply change the active.ident using the `Idents()` or set a different `group.by` parameter if you want to visualize different.  But you never want to remove meta data from an analysis.

Best,
Sam
